lib/iotagentCore.js

```diff
@@ -52,9 +52,12 @@ function deviceProvisioningMiddleware(device, callback) {
         if (!mappingFound) {
             config.getLogger().error(context, 'device provisioning mapping not found');
             callback(new errors.DataMappingNotFound(device.type), device);
+        } else {
+            callback(null, device);
         }
+    } else {
+        callback(null, device);
     }
-    callback(null, device);
 }
```

Comment:
This is an existing bug. Following the program flow, sometimes the callback could be called twice. The modifications done in this PR avoid this.

This has passed unnoticed until now, but after upgrading the driver version maybe do to some timing change the bug has arisen.

NTC (informative)